nntrainer/models/neuralnet.cpp

```diff
@@ -381,22 +375,13 @@ void NeuralNetwork::load(const std::string &file_path,
 
     auto model_file = checkedOpenStream<std::ifstream>(
       file_path, std::ios::in | std::ios::binary);
-    if (!loadedWeight) {
-      for (auto iter = model_graph.cbegin(); iter != model_graph.cend();
-           iter++) {
-        (*iter)->read(model_file);
-      }
-      loadedWeight = true;
-      bin_file_pos = model_file.tellg();
-      load_path = file_path;
-      return;
+    for (auto iter = model_graph.cbegin(); iter != model_graph.cend(); iter++) {
+      (*iter)->read(model_file);
     }
     try {
       /// this is assuming that the failure is allowed at the end of the file
       /// read. so, after this line, additional read shouldn't be called
-      model_file.seekg(bin_file_pos);
-
-      if (istrequal(opt->getType(), "adam")) {
+      if (opt && istrequal(opt->getType(), "adam")) {
```

Comment:
opt is sgd.. and if the file is saved with adam.. then, I guess we need to skip to read opt adam variables.. and try to read iteration number and epoch.

Reply:
Yes, it can be solved by changing file format (eg, reading iteration and epoch before optimizer section) for now, the patch is done in a way not breaking the file format. I guess the strategy in #1851 this can be fixed although it is reverted for other reason.

Reply:
this is fine for now but can be done lazily with #1849, where optimizer will be determined at training time and not at initialize.